src/test/java/org/datadog/jmxfetch/JMXServerSupervisorClient.java

```diff
@@ -0,0 +1,81 @@
+package org.datadog.jmxfetch;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Map;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class JMXServerSupervisorClient {
+    private final String host;
+    private final int port;
+
+    public JMXServerSupervisorClient(String host, int port) {
+        this.host = host;
+        this.port = port;
+    }
+
+    public void setRmiHostname(String rmiHostname) throws IOException {
+        sendPostRequestWithFormParams("/config", Collections.singletonMap("rmiHostname", rmiHostname));
+    }
+
+    private void sendPostRequestWithFormParams(String endpoint, Map<String, String> formParams) throws IOException {
```

Comment:
consolidate this into json body, don't duplicate the post request sender with one version for json body and a second for form params.